ciao-controller/types/types.go

```diff
@@ -714,3 +716,34 @@ type QuotaUpdateRequest struct {
 type QuotaListResponse struct {
 	Quotas []QuotaDetails `json:"quotas"`
 }
+
+func NewTenantHardwareAddr(ip net.IP) net.HardwareAddr {
```

Comment:
I don't think types.go is the right place. Currently only the only "code" in this file for manipulating the types defined in it. I think putting it in the datastore package as datastore/utils.go or putting a utils.go in the ciao-controller main package would be better.